include/dd/Package.hpp

```diff
@@ -484,6 +484,52 @@ template <class Config> class Package {
     return f;
   }
 
+  // generate general GHZ state with n qubits
+  vEdge makeGHZState(const std::size_t n, const size_t start = 0) {
+    if (n + start > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n + start) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    std::vector<BasisStates> oneBasisState =
+        std::vector<BasisStates>(n, BasisStates::one);
+    auto zeroState = makeZeroState(n);
+    auto oneState = makeBasisState(n, oneBasisState);
+    auto f = normalize(add(zeroState, oneState), false);
+
+    return f;
+  }
+
+  // generate general W state with n qubits
+  vEdge makeWState(const std::size_t n, const size_t start = 0) {
+    if (n + start > nqubits) {
+      throw std::runtime_error{
+          "Requested state with " + std::to_string(n + start) +
+          " qubits, but current package configuration only supports up to " +
+          std::to_string(nqubits) +
+          " qubits. Please allocate a larger package instance."};
+    }
+
+    if (n < 2) {
+      return makeBasisState(n, {BasisStates::one});
+    }
+
+    auto f = vEdge::one;
+    for (size_t i = 0; i < n + start; ++i) {
+      std::vector<BasisStates> state =
+          std::vector<BasisStates>(n, BasisStates::zero);
+      state[i] = BasisStates::one;
+
+      auto newBasisState = makeBasisState(n, state);
+      f = add(newBasisState, f);
+    }
+    f = normalize(f, false);
+    return f;
+  }
```

Comment:
I created some examples by hand and some using [https://www.cda.cit.tum.de/app/ddvis/](MQT DDVis) and tried to do the same as before (i.e. create the left and right subtree in parallel). The only note here is that the first and the last nodes must be created separately outside the loop. 

I know that having 3 variables might be much, however, I tried keeping it as simple as possible so that it can be readable and easier to understand. Lastly, as far as I know, the normalization step should also be linear in the number of qubits, right?

Reply:
See also the other comment from the review just now. Your implementation looked pretty reasonable already. Just a couple of further improvements.

Normalisation actually happens on every call to `makeDDNode` and only touches the node it is called on and the edge weights to its successors. So it is actually O(1).